normalizer/eval_utils.py

```diff
@@ -117,34 +152,56 @@ def parse_filepath(fp: str):
         references = [datum["text"] for datum in manifest]
         predictions = [datum["pred_text"] for datum in manifest]
 
+        time = [datum["time"] for datum in manifest]
+        duration = [datum["duration"] for datum in manifest]
+        compute_rtfx = all(time) and all(duration)
+
         wer = wer_metric.compute(references=references, predictions=predictions)
         wer = round(100 * wer, 2)
 
+        if compute_rtfx:
+            rtfx = sum(duration) / sum(time)
+            rtfx = round(rtfx, 4)
+        else:
+            rtfx = None
+
         result_key = f"{model_id_of_file} | {dataset_id}"
-        results[result_key] = wer
+        results[result_key] = {"wer": wer, "rtfx": rtfx}
 
     print("*" * 80)
     print("Results per dataset:")
     print("*" * 80)
 
     for k, v in results.items():
-        print(f"{k}: WER = {v:0.2f} %")
+        metrics = f"{k}: WER = {v['wer']:0.2f} %"
+        if v["rtfx"] is not None:
+            metrics += f", RTFX = {v['rtfx']:0.2f}"
+        print(metrics)
 
     # composite WER should be computed over all datasets and with the same key
     composite_wer = defaultdict(float)
+    composite_rtfx = defaultdict(float)
     count_entries = defaultdict(int)
     for k, v in results.items():
         key = k.split("|")[0].strip()
-        composite_wer[key] += v
+        composite_wer[key] += v["wer"]
+        if v["rtfx"] is not None:
+            composite_rtfx[key] += v["rtfx"]
+        else:
+            composite_rtfx[key] = None
         count_entries[key] += 1
 
     # normalize scores & print
     print()
     print("*" * 80)
-    print("Composite WER:")
+    print("Composite Results:")
     print("*" * 80)
     for k, v in composite_wer.items():
         wer = v / count_entries[k]
```

Comment:
Just a note, but it looks like you are computing WER and RTFx in an unweighted way, which is normally not considered correct. Presumably that is okay, since we don't want a massive test set (like Gigaspeech's 100 hour test set) to outweigh a smaller, different dataset. Anyway, I just wanted to point it out.

Reply:
Indeed - I keep the RTFx calculation consistent with the un-weighted WER that we currently use. For RTFx, it would indeed be better to switch to a weighted computation. For WER, it's more open for debate: we typically see both un-weighted and weighted WERs reported in papers. Perhaps for consistency with the current benchmark WERs, week keep it un-weighted?